res/controllers/Icon-iControls-scripts.js

```diff
@@ -0,0 +1,187 @@
+    const mapIndexToChannel = function(index) {
+        switch (Math.abs(index) % 4) {
+        case 0: return 3;
+        case 1: return 1;
+        case 2: return 2;
+        case 3: return 4;
+        }
+    };
+
+    class Deck extends components.Deck {
+        constructor() {
+            super([1, 2, 3, 4]);
+            // Transport buttons
+            this.playButton = new components.PlayButton({
+                group: "[Channel1]",
+                midi: [0x90, 0x5E], // Play transport
+                type: components.Button.prototype.types.toggle,
+            });
+            this.cueButton = new components.CueButton({
+                group: "[Channel1]",
+                midi: [0x90, 0x5D], // Pause transport
+                type: components.Button.prototype.types.push,
+            });
+            this.backButton = new components.Button({
+                group: "[Channel1]",
+                midi: [0x90, 0x5B],
+                key: "beatjump_backward",
+            });
+            this.forwardButton = new components.Button({
+                group: "[Channel1]",
+                midi: [0x90, 0x5C],
+                key: "beatjump_forward",
+            });
+            this.loopButton = new components.Button({
+                group: "[Channel1]",
+                midi: [0x90, 0x56],
+                inKey: "beatloop_activate",
+                outKey: "loop_enabled",
+            });
+
+            this.leftDeckBtn = new components.Button({
+                group: "[Channel1]",
+                midi: [0x90, 0x30],
+                type: components.Button.prototype.types.powerWindow,
+                deck: this,
+                inToggle: function() {
+                    if (this.isLongPressed) {
+                        this.deck.setCurrentDeck("[Channel3]");
+                    } else {
+                        this.deck.setCurrentDeck("[Channel1]");
+                    }
+                },
+            });
+            this.rightDeckBtn = new components.Button({
+                group: "[Channel2]",
+                midi: [0x90, 0x31],
+                type: components.Button.prototype.types.powerWindow,
+                deck: this,
+                inToggle: function() {
+                    if (this.isLongPressed) {
+                        this.deck.setCurrentDeck("[Channel4]");
+                    } else {
+                        this.deck.setCurrentDeck("[Channel2]");
+                    }
+                },
+            });
+        }
+    }
+
+    class VelocityPot extends components.Pot {
+        constructor(params) {
+            super(Object.assign({
+                softTakeover: false,
+                relative: true,
+            }, params));
+        }
+        inValueScale(value) {
+            if (value & 0x40) {
+                return this.inGetParameter() - ((value & ~0x40) / 100);
+            } else {
+                return this.inGetParameter() + (value / 100);
+            }
```

Comment:
isn't this just the usual twos-complement thing?
Since numbers in JS are always floats, binary operations are usually slow, its better to just do the arithmatic equivalent instead:
```suggestion
            value = value > 0x40 ? value - 0x80 : value;
            return this.inGetParameter() + (value / 100);
```